### PR TITLE
Add Tests Framework

### PR DESCRIPTION
**Pull request by mx4492**
I intend to add some functionality to the script, so as preparation I am adding a minimal test suite.

### CODE REVIEW

**Comment by soh335** [comment]
thanks for p-r.  I have 2 questions. 
- Why use rakefile ? I'm not good at node, coffeescript and hubot. But rakefile seems not to be need in this project.
- What functionality do you want to add ? I want to keep simple functionality and same interface to remind.coffee.

**Comment by mx4492** [comment]
> Why use rakefile ? I'm not good at node, coffeescript and hubot. But rakefile seems not to be need in this project.

I am used to working with Rake, which is why I added it. But you are right, it is unnecessary in this project, so I have removed it (see latest commit).

> What functionality do you want to add ? I want to keep simple functionality and same interface to remind.coffee.

I want to add support for less strict parsing (similar to https://github.com/soh335/hubot-remind-at/pull/1) and support for relative times (similar to what is already provided by remind.coffee, but a bit enhanced). Since I see that you want to keep the same interface as remind.coffee, I can do that on a separate fork and not modify this project. 

But still, having a test suite will be useful to you in the future so you might consider merging at least this pull request :) Other functionality I will add in my own project.

**Comment by soh335** [comment]
of course test suite is welcome :ok_woman:

**Comment by soh335** [comment]
timezone problem is exist. 
Hubot response message ( `I'll remind you to ... at xxx` ) is changed by timezone. So we should fixed timezone. please add `process.env.TZ = 'UTC'` to first line at test/remind.coffee and change `GMT+0530 (IST)` to `GMT+0000 (UTC)`

File: test/remindat.coffee [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+helper = new Helper('../scripts')
+
+describe 'Reminder formats', ->
+  room = null
+  beforeEach ->
+    room = helper.createRoom()
+    room.lastMessageText = ->
+      [...,last] = room.messages
+      last[1]
+
```
Comment (soh335) [inline test/remindat.coffee:13]: I'd prefer not to define new method for testing to object.

how about is ? 

```
assert.deepEqual ["hubot", "I'll remind you to do task at Thu Aug 17 2513 12:00:00 GMT+0000 (UTC)"], room.messages[1]
```

it's simple.

**Comment by mx4492** [comment]
Updated.

**Comment by soh335** [comment]
thanks @mx4492 ! :+1:

**Pull request merged**